test/rubygems/test_gem_package.rb

```diff
@@ -738,6 +738,7 @@ def test_install_location_extra_slash
     file.taint if RUBY_VERSION < '2.7'
 
     destination = @destination.sub '/', '//'
+    destination = ?/ + destination if destination.start_with? '//'
```

Comment:
But now this test always tests with 3 slashes, no? Because the previous line will always return a path with two leading slashes?

Can we instead detect that the test is running on Alpine and add the extra slash just in that case, with a comment? I think it would make more explicit what's going on.

Reply:
I'm pretty sure it does indeed not matter. But we'll be changing the test to test something different from the original, so I think we should at least double check the original issue that led into adding the test, and make sure that using 3 double slashes covers the original issue just as fine.

Even if the commit message is well detailed, I think adding a comment may prevent someone from breaking the test in the future thinking two slashes are enough 😅, given that we don't really test on Alpine or any other musl libc distribution. But if you don't feel it's needed, it's fine with me.

Also, regarding `?/`, today I learned. Can we use something more commonly used like `'/'`?

Reply:
I was checking the code and it was introduced to fix https://github.com/rubygems/rubygems/issues/508. In that issue, leading extra slashes were not the issue, it was double slashes inside gem paths, and I don't think that's a thing in real life either. I guess the original issue came from something RVM was doing when merging paths that resulted in duplicated `//` in the middle of `GEM_HOME` or something.

So maybe the easiest thing to do here is to remove the line above and pass `@destination` directly to `#install_location`. That has no chances of being reverted, needs no comment, and the test is still checking normalization of inner double slashes which seems to work consistently across all libc implementations.

Reply:
To be clear, this would be my suggested alternative patch

```diff
diff --git a/test/rubygems/test_gem_package.rb b/test/rubygems/test_gem_package.rb
index 34660bdccd..e3dce87c95 100644
--- a/test/rubygems/test_gem_package.rb
+++ b/test/rubygems/test_gem_package.rb
@@ -737,9 +737,7 @@ def test_install_location_extra_slash
     file = 'foo//file.rb'.dup
     file.taint if RUBY_VERSION < '2.7'
 
-    destination = @destination.sub '/', '//'
-
-    destination = package.install_location file, destination
+    destination = package.install_location file, @destination
 
     assert_equal File.join(@destination, 'foo', 'file.rb'), destination
     refute destination.tainted? if RUBY_VERSION < '2.7'
```

Reply:
> Note: Technically we could replace last / with //, but not sure it's worth the effort. We are basically testing realpath at that point.

Yes, I was exactly looking into that, I was thinking of `@destination.reverse.sub('/', '//').reverse`, but then I realized that the test didn't really need this.